### PR TITLE
[Uptime] Add docs for autodiscover hints

### PR DESCRIPTION
**Pull request by andrewvc**
Autodiscover hints were introduced in #13119 but not documented. This adds the docs.

Resolves https://github.com/elastic/beats/issues/14245

### CODE REVIEW

File: heartbeat/docs/autodiscover-hints.asciidoc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+```
+
+To configure this with autodiscover add the following line to your Dockerfile to add the requisite labels.
+
+```
+LABEL co.elastic.monitor/1.type=tcp co.elastic.monitor/1.hosts='${data.host}:6379' co.elastic.monitor/1.schedule='@every 10s'
+LABEL co.elastic.monitor/2.type=icmp co.elastic.monitor/2.hosts='${data.host}' co.elastic.monitor/2.schedule='@every 10s'
+```
+
+Then, add the following to `heartbeat.yml:
```
Comment (justinkambic) [inline heartbeat/docs/autodiscover-hints.asciidoc]:
```suggestion
Then, add the following to `heartbeat.yml`:
```

File: heartbeat/docs/autodiscover-hints.asciidoc [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+the container starts, {beatname_uc} will check if it contains any hints and launch the proper config for
+it. Hints tell {beatname_uc} how to get logs for the given container.
+
+Hints for Heartbeat take the form `co.elastic.monitor/MONITOR_INDEX.CONFIG_OPTION=VALUE`. Where `MONITOR_INDEX` represents
+the index of monitor, `CONFIG_OPTION` is any of the <<configuration-heartbeat-options>> options, with `VALUE` being the value
+you'd like to set that option to.
+
+As an example let's convert the regular syntax heartbeat monitors below into autodiscover hints.
+
+```yaml
```
Comment (dedemorton) [inline heartbeat/docs/autodiscover-hints.asciidoc]: Use asciidoctor source code block formatting with syntax highlighting here (what you have here is Markdown). See https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#source-code

File: heartbeat/docs/autodiscover-hints.asciidoc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+To configure this with autodiscover add the following line to your Dockerfile to add the requisite labels.
+
+```
+LABEL co.elastic.monitor/1.type=tcp co.elastic.monitor/1.hosts='${data.host}:6379' co.elastic.monitor/1.schedule='@every 10s'
+LABEL co.elastic.monitor/2.type=icmp co.elastic.monitor/2.hosts='${data.host}' co.elastic.monitor/2.schedule='@every 10s'
+```
+
+Then, add the following to `heartbeat.yml:
+
+```yaml
```
Comment (dedemorton) [inline heartbeat/docs/autodiscover-hints.asciidoc]: same comment as above

File: heartbeat/docs/autodiscover-hints.asciidoc [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+heartbeat.autodiscover:
+  providers:
+  - type: docker
+    hints.enabled: true
+```
+
+Then, start {{beatname_uc}}. This should discover the container you built and ping it once with ICMP, another time with TCP.
```
Comment (dedemorton) [inline heartbeat/docs/autodiscover-hints.asciidoc]: Did you mean to use double curly braces here?

File: heartbeat/docs/autodiscover-hints.asciidoc [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+For a full list of variables available see the provider specific docs in <<configuration-autodiscover>>.
```
Comment (dedemorton) [inline heartbeat/docs/autodiscover-hints.asciidoc]:
```suggestion
For a full list of variables available, see the provider-specific docs in <<configuration-autodiscover>>.
```

File: heartbeat/docs/autodiscover-hints.asciidoc (hunk 4 above)
Comment (andrewvc) [reply heartbeat/docs/autodiscover-hints.asciidoc]: Nope! Fixed

**Comment by andrewvc** [comment]
@dedemorton I've made all the changes you requested and tested locally.

**Comment by andrewvc** [comment]
@jenkins retest this please

File: heartbeat/docs/autodiscover-hints.asciidoc [hunk 6]
```diff
@@ -0,0 +1,42 @@
+{beatname_uc} supports autodiscover based on hints from the both Docker and Kubernetes. The hints system looks for
+hints in Kubernetes Pod annotations or Docker labels that have the prefix `co.elastic.monitor`. As soon as
+the container starts, {beatname_uc} will check if it contains any hints and launch the proper config for
+it. Hints tell {beatname_uc} how to get logs for the given container.
+
+Hints for Heartbeat take the form `co.elastic.monitor/MONITOR_INDEX.CONFIG_OPTION=VALUE`. Where `MONITOR_INDEX` represents
```
Comment (dedemorton) [inline heartbeat/docs/autodiscover-hints.asciidoc]:
```suggestion
Hints for {beatname_uc} take the form `co.elastic.monitor/MONITOR_INDEX.CONFIG_OPTION=VALUE`, where `MONITOR_INDEX` represents
```

File: heartbeat/docs/autodiscover-hints.asciidoc [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+the index of monitor, `CONFIG_OPTION` is any of the <<configuration-heartbeat-options>> options, with `VALUE` being the value
```
Comment (dedemorton) [inline heartbeat/docs/autodiscover-hints.asciidoc]:
```suggestion
the index of monitor, and `CONFIG_OPTION` is any of the <<configuration-heartbeat-options>> options, with `VALUE` being the value
```

File: heartbeat/docs/autodiscover-hints.asciidoc [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+you'd like to set that option to.
+
+As an example let's convert the regular syntax heartbeat monitors below into autodiscover hints.
```
Comment (dedemorton) [inline heartbeat/docs/autodiscover-hints.asciidoc]:
```suggestion
As an example let's convert the regular syntax {beatname_uc} monitors below into autodiscover hints.
```

File: heartbeat/docs/autodiscover-hints.asciidoc [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+-- heartbeat.monitors:
+- type: tcp
+  hosts: '${data.host}:6379'
+  schedule: @every 10s
+- type: icmp
+  hosts: '${data.host}'
+  schedule: @every 10s
+----------------------------------------------------------------------
+
+To configure this with autodiscover add the following line to your Dockerfile to add the requisite labels.
```
Comment (dedemorton) [inline heartbeat/docs/autodiscover-hints.asciidoc]:
```suggestion
To configure this with autodiscover, add the following line to your Dockerfile to add the requisite labels.
```

**Pull request merged**